MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -1101,6 +1101,25 @@
 	return rslt;
 }
 
+// Don't need to do this for Apple GPUs, where the GPU and CPU timestamps
+// are the same, or if we're not using GPU timestamp counters.
+void MVKPhysicalDevice::startTimestampCorrelation(MTLTimestamp& cpuStart, MTLTimestamp& gpuStart) {
+	if (_properties.vendorID == kAppleVendorId || !_timestampMTLCounterSet) { return; }
+	[_mtlDevice sampleTimestamps: &cpuStart gpuTimestamp: &gpuStart];
```

Comment:
You also have to check the OS version before using this method. I don't think it's supported prior to Catalina.

Reply:
> You also have to check the OS version before using this method. I don't think it's supported prior to Catalina.

That should be covered by the `!_timestampMTLCounterSet` guard code. Both `counterSamplingPoints` and `_timestampMTLCounterSet` will be empty prior to 10.15.

Reply:
Ah, I see. All right, then.